### ISSUE TITLE
arguments during verification don't match arguments passed to actual mock invocation

### ISSUE THREAD

**Issue by karol-swiniarski-zandware** [mockito/mockito#3082]
i see in `InterceptedInvocation` constructor that when intercepting calls on mock objects, arguments are saved directly as passed to method call not a copy/snapshot of this values.

since only shallow copy is made, this leads to situation, that intercepted call record can be altered by changing content of passed arguments (unluess they are all immutable) and during verification we verify calls agains altered arguments, not that were actually passed during execution

in my opinion, this is a severe issue

i know i can intercept calls and write copy of actual arguments which i can later use for verification, but it seems like someting that could be handled by framework by default (or at least an option)

**Comment by karol-swiniarski-zandware** [comment]
is it possible to create branch with possible solution to this issue?

**Comment by karol-swiniarski-zandware** [comment]
bump

**Comment by LeMikaelF** [comment]
This looks like a duplicate of https://github.com/mockito/mockito/issues/1153.

**Comment by karol-swiniarski-zandware** [comment]
yes looks like it. and the one you mentioned is still opened as i see
i also see that something called `ClonesArguments` is mentioned in the topic - maybe this is solution
